### PR TITLE
Ignorable tables

### LINKED ISSUE #139: Add option to ignore tables completely
Sometimes there are tables in the schema that you don't want included in anything: Not orm.py, not ssg.py, not at all.

This came up when working on some OMOP data.

### PR DESCRIPTION
**Pull request by mhauru**
Add option to ignore tables. In so doing, also rework example_orm.py.
  * Add option in the config to ignore some table entirely
  * Start checking in the functional tests that the produced ORM is as expected. This allows testing the above feature.
  * For this purpose, completely rework example_orm.py so that it is now the valid output of make-tables on src.dump.
  * Make make-tables run black on the outputted code.

Closes #139

### CODE REVIEW

**Comment by mhauru** [comment]
The tests pass for me locally. @Iain-S was this the same issue you already fixed in some other PR?

**Comment by mhauru** [comment]
I ran into an issue where I some tables kept getting included in `orm.py` even though I set `ignore: true`. This was because one of the not-ignored tables had a foreign key reference to it. That makes sense, you then have to have that table in your `orm.py`, but we may still want to ignore that table for the purposes of creating generators, data, etc. I added a commit that implements this functionality.

File: sqlsynthgen/make.py [hunk 1]
```diff
@@ -440,15 +462,39 @@ def _get_generator_for_vocabulary_table(
...
+        table_config = tables_config.get(table_name, {})
+        ignore = table_config.get("ignore", False)
+        if ignore:
+            logging.warning(
+                "Table %s is supposed to be ignored but there is a foreign key "
+                "reference to it. "
+                "You may need to create this table manually at the dst schema before "
+                "running create-tables.",
+                table_name,
+            )
```
Comment (Iain-S) [inline sqlsynthgen/make.py:497]: Is it the case that, even after `refect()`, metadata may still contain some of the `ignored` tables (if they are referenced by a foreign key) so we need to check and warn?

**Review by Iain-S — APPROVED** [review]
All looks good to me. I've pushed a couple of minor changes straight to this branch.

**Pull request merged**